packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -44,12 +44,11 @@ export default ({ strapi }: { strapi: Strapi }) => ({
   },
 
   async getAvailableStatus(uid: Common.UID.ContentType, document: DocumentVersionSelector) {
-    if (!document.locale) return null;
-
     // Find if the other status of the document is available
     const otherStatus = document.status === 'published' ? 'draft' : 'published';
 
     return strapi.documents(uid).findOne(document.id, {
+      // TODO: Do not filter by locale if i18n is disabled
       locale: document.locale,
```

Comment:
does it not default if `undefined`?

Reply:
it defaults to NULL when i18n is not enabled, but indeed even if  `document.locale` is undefined it should work correctly.
I just left the todo to really be sure we dont have an issue with the filter